src/globus_cli/commands/endpoint/permission/_common.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+from globus_cli.termio import formatters
+
+
+class AclPrincipalFormatter(formatters.auth.PrincipalDictFormatter):
+    # customize the formatter to provide the `principal_type` as the fallback value for
+    # unrecognized types. This handles various cases in which
+    # `principal_type=all_authenticated_users` or similar, which is the shape of the
+    # data from Globus Transfer
+    def fallback_rendering(self, principal: str, principal_type: str):
+        return principal_type
+
+    def render_group_id(self, group_id: str) -> str:
+        return f"https://app.globus.org/groups/{group_id}"
```

Comment:
This is something I've done to match the current behavior, which is not environment-specific and just uses this format string. I wanted to minimize breakage and "unrelated" changes.

I've encoded what I _think_ is the preferable output (`"Globus Group ({group_id})"`) into the base class. But this is meant to be something we discuss further.
By surfacing this formatting ugliness into a more readable location, I'm hoping we can setup some future conversation to decide on how we want this to render. Maybe we should change the base class to use these URLs and update the outputs which use the other format string.

I would be happy to add a comment here with a TODO if you're concerned about the tenuousness of this formatting being forgotten.

Reply:
If this is a URL that may change by environment, rather than a URI with a known prefix, a TODO might be useful as a call-out for future readers (including myself!).

Reply:
TODOs are now in place from https://github.com/globus/globus-cli/pull/692/commits/8a46b679fb2bebd60b7de560673fd28dbd49f722